tests/unit/amo/sagas/test_blocks.js

```diff
@@ -0,0 +1,101 @@
+import SagaTester from 'redux-saga-tester';
+
+import * as blocksApi from 'amo/api/blocks';
+import blocksReducer, {
+  abortFetchBlock,
+  fetchBlock,
+  loadBlock,
+} from 'amo/reducers/blocks';
+import blocksSaga from 'amo/sagas/blocks';
+import { createApiError } from 'core/api';
+import apiReducer from 'core/reducers/api';
+import {
+  createFakeBlockResult,
+  createStubErrorHandler,
+  dispatchClientMetadata,
+} from 'tests/unit/helpers';
+
+describe(__filename, () => {
+  let errorHandler;
+  let mockBlocksApi;
+  let sagaTester;
+
+  beforeEach(() => {
+    errorHandler = createStubErrorHandler();
+    mockBlocksApi = sinon.mock(blocksApi);
+    sagaTester = new SagaTester({
+      initialState: dispatchClientMetadata().state,
+      reducers: {
+        api: apiReducer,
+        blocks: blocksReducer,
+      },
+    });
+    sagaTester.start(blocksSaga);
+  });
+
+  describe('fetchBlock', () => {
+    function _fetchBlock(params) {
+      sagaTester.dispatch(
+        fetchBlock({
+          errorHandlerId: errorHandler.id,
+          guid: 'some-guid',
+          ...params,
+        }),
+      );
+    }
+
+    it('calls the API to fetch a block by GUID', async () => {
```

Comment:
Do we need one that works on block pk, or is it just assumed it'll work because whatever the "guid" is provided is just passed onto the API?

Reply:
This:

> whatever the "guid" is provided is just passed onto the API